File: dsp07.pas
(********  PROGRAM NO.7   DISCRETE-TIME FOURIER SERIES  ********)
(**************  FOR SIGNAL WITH 7 SAMPLE VALUES  **************)
program DSP07;
var X: array[1..7] of real;
var AR: array[1..7] of real;
var AI: array[1..7] of real;
var N,K: integer;
begin
  for N:=1 to 7 do
    begin
      AR[N]:=0; AI[N]:=0;
    end;
  X[1]:=2; X[2]:=1; X[3]:=-2; X[4]:=3;
  X[5]:=-1; X[6]:=-1; X[7]:=1;
  for K:=1 to 7 do
    begin
      for N:=1 to 7 do
        begin
          AR[K]:=AR[K]+(1/7)*X[N]*cos(2*3.141593*(K-1)*(N-1)/7);
          AI[K]:=AI[K]-(1/7)*X[N]*sin(2*3.141593*(K-1)*(N-1)/7);
        end;
    end;
  for N:=1 to 7 do
    begin
      writeln(AR[N],AI[N]);
    end;
end.
(***************************************************************)

File: dsp07.c
#include <graph.h>
#include <conio.h>
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

float   fX[] = {2, 1, -2, 3, -1, -1, 1}, fAR[7], fAI[7];

void main (void)
    {
    /*
     *********  PROGRAM NO.7   DISCRETE-TIME FOURIER  **********
     *********  SERIES FOR SIGNAL WITH 7 SAMPLE VALUES  ********
     */
    int    iCount, iCount2;

    _clearscreen  (_GCLEARSCREEN);
    printf ("PROGRAM 7: Discrete Fourier Series\n");
    printf ("for real signal with 7 sample values\n\n");

    for (iCount2 = 1; iCount2 < 8; iCount2++)
      for (iCount = 1; iCount < 8; iCount++)
        {
        fAR[iCount2-1] = fAR[iCount2-1] + (1.0/7.0) *
              fX[iCount-1] * cos (2.0 * 3.141593 *
              (float)(iCount2-1) * (float)(iCount-1) / 7.0);
        fAI[iCount2-1] = fAI[iCount2-1] - (1.0/7.0) *
              fX[iCount-1] * sin (2.0 * 3.141593 *
              (float)(iCount2-1) * (float)(iCount-1) / 7.0);
        }

    for (iCount2 = 0; iCount2 < 7; iCount2++)
        printf ("%d     % f, % f\n", iCount2,
        fAR[iCount2], fAI[iCount2]);

    /*
     ***********************************************************
     */
    }
